src/game_map.cpp

```diff
@@ -572,202 +572,226 @@ static CollisionResult TestCollisionDuringMove(
 	return NoCollision;
 }
 
-bool Game_Map::MakeWay(int x, int y, int d, const Game_Character& self) {
-	int new_x = RoundX(x + (d == Game_Character::Right ? 1 : d == Game_Character::Left ? -1 : 0));
-	int new_y = RoundY(y + (d == Game_Character::Down ? 1 : d == Game_Character::Up ? -1 : 0));
+static int GetPassableMask(int old_x, int old_y, int new_x, int new_y) {
+	int bit = 0;
+	if (new_x > old_x) { bit |= Passable::Right; }
+	if (new_x < old_x) { bit |= Passable::Left; }
+	if (new_y > old_y) { bit |= Passable::Down; }
+	if (new_y < old_y) { bit |= Passable::Up; }
+	return bit;
+}
 
-	if (!Game_Map::IsValid(new_x, new_y))
+static bool WouldCollide(const Game_Character& self, const Game_Character& other, bool self_conflict) {
+	if (self.GetThrough() || other.GetThrough()) {
 		return false;
+	}
 
-	if (self.GetThrough()) return true;
-
-	// A character can move to a position with an impassable tile by
-	// standing on top of an event below it. These flags track whether
-	// we stepped off an event and therefore don't need to check the
-	// passability of the tile layer below.
-	bool stepped_off_event = false;
-	bool stepped_onto_event = false;
-
-	for (Game_Event& other : GetEvents()) {
-		CollisionResult result = TestCollisionDuringMove(x, y, new_x, new_y, d, self, other);
```

Comment:
Seems that TestCollisionDuringMove and the CollisionResult enum are now dead code and can be removed?

Reply:
Removed